src/service/nickname.ts

```diff
@@ -0,0 +1,89 @@
+import {
+    Collection,
+    GuildMember,
+    OAuth2Guild,
+    PartialGuildMember,
+} from "discord.js";
+import { getUser } from "../wrapper/wrapper";
+import { config } from "../config";
+
+import { rankMap } from "../service/util/constants";
+import { UserRankRepository } from "../repository/UserRankRepository";
+import { UserRank } from "../entity/UserRank";
+import { UserData } from "../wrapper/models/user";
+import { Repository } from "typeorm";
+import { makeLogger } from "../logger";
+
+const logger = makeLogger(module);
+
+export async function processUser(
+    u: UserRank,
+    guilds: Collection<string, OAuth2Guild>
+) {
+    try {
+        const user = await getUser(config.apiKey, u.discord_id);
+        if ((guilds && user.name != u.user_name) || user.rank != u.rank) {
+            u.guild_ids.forEach(async (g) => {
```

Comment:
this is not `await`ed again, which means that any errors happening in this loop will not be caught, and just quietly disappear